core/services/chainlink/config_general.go

```diff
@@ -196,11 +199,30 @@ func (g *generalConfig) StarknetConfigs() starknet.StarknetConfigs {
 }
 
 func (g *generalConfig) Validate() error {
-	_, err := utils.MultiErrorList(multierr.Combine(
+	return g.validate()
+}
+
+func (g *generalConfig) validate(secrets ...v2.Validated) error {
+	err := multierr.Combine(
 		validateEnv(),
-		g.c.Validate(),
-		g.secrets.Validate()))
-	return err
+		g.c.Validate())
+
+	if len(secrets) == 0 {
+		err = multierr.Append(err, g.secrets.Validate())
+	} else {
+		serrs := []error{err}
+		for _, s := range secrets {
+			serrs = append(serrs, s.ValidateConfig())
+		}
+		err = multierr.Combine(serrs...)
```

Comment:
I appending directly is equivalent, no?
```suggestion
		for _, s := range secrets {
			errs = multierr.Append(err, s.ValidateConfig())
		}
```